hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsAsyncQueueService.kt

```diff
@@ -49,44 +51,41 @@ open class HmppsAsyncQueueService(
       .map { retryDlqRequest -> retryDlqMessages(retryDlqRequest) }
 
   private suspend fun HmppsAsyncQueue.retryDlqMessages(): RetryDlqResult {
-    if (sqsAsyncDlqClient == null || dlqUrl == null) return RetryDlqResult(0, mutableListOf())
+    if (sqsAsyncDlqClient == null || dlqUrl == null) return RetryDlqResult(0, listOf())
     val messageCount = sqsAsyncDlqClient.countMessagesOnQueue(dlqUrl!!)
-    val messages = mutableListOf<Message.Builder>()
-    repeat(messageCount) {
-      sqsAsyncDlqClient.receiveMessage(ReceiveMessageRequest.builder().queueUrl(dlqUrl).maxNumberOfMessages(1).attributeNames(QueueAttributeName.ALL).build())
-        .thenApply { it.messages().firstOrNull() }
-        .thenApply { msg ->
-          msg?.run {
-            sqsAsyncClient.sendMessage(SendMessageRequest.builder().queueUrl(queueUrl).messageBody(msg.body()).messageAttributes(msg.messageAttributes()).build())
-            sqsAsyncDlqClient.deleteMessage(DeleteMessageRequest.builder().queueUrl(dlqUrl).receiptHandle(msg.receiptHandle()).build())
-            messages += msg.toBuilder()
-          }
-        }
-    }
+    val map: Map<String, Any> = HashMap()
+
+    val messages = (1..messageCount)
+      .map { sqsAsyncDlqClient.receiveMessage(ReceiveMessageRequest.builder().queueUrl(dlqUrl).maxNumberOfMessages(1).attributeNames(QueueAttributeName.ALL).build()).await() }
+      .mapNotNull { it.messages().firstOrNull() }
+      .map { msg ->
+        sqsAsyncClient.sendMessage(SendMessageRequest.builder().queueUrl(queueUrl).messageBody(msg.body()).messageAttributes(msg.messageAttributes()).build())
```

Comment:
just be aware, if either of these fail you will not know. Might be worth experimenting with adding `.exceptionally`  or something to log errors even if you don't return a 500 error or whatever